pkg/server/node.go

```diff
@@ -1044,3 +1045,81 @@ func (n *Node) RangeFeed(
 	}
 	return nil
 }
+
+func (n *Node) UnsafeHealRange(
+	ctx context.Context, req *roachpb.UnsafeHealRangeRequest,
+) (_ *roachpb.UnsafeHealRangeResponse, rErr error) {
+	defer func() {
+		if r := recover(); r != nil {
+			rErr = errors.Errorf("%v", r)
+		}
+	}()
+
+	var expValue *roachpb.Value
+	var desc roachpb.RangeDescriptor
+	if err := n.storeCfg.DB.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
+		kvs, err := sql.ScanMetaKVs(ctx, txn, roachpb.Span{
+			// TODO(tbg): paginate.
+			Key:    roachpb.KeyMin,
+			EndKey: roachpb.KeyMax,
+		})
+		if err != nil {
+			return err
+		}
+
+		for i := range kvs {
+			if err := kvs[i].Value.GetProto(&desc); err != nil {
+				return err
+			}
+			if desc.RangeID == req.RangeID {
+				expValue = kvs[i].Value
+				return nil
+			}
+		}
+		return errors.Errorf("r%d not found", req.RangeID)
+	}); err != nil {
+		return nil, err
+	}
+
+	var s *storage.Store
+	if err := n.stores.VisitStores(func(inner *storage.Store) error {
+		if s == nil {
+			s = inner
+		}
+		return nil
+	}); err != nil {
+		return nil, err
+	}
+	storeID, nodeID := s.Ident.StoreID, s.Ident.NodeID
+
+	dead := append([]roachpb.ReplicaDescriptor(nil), desc.Replicas().All()...)
```

Comment:
Note to self: Here we're just constructing the range descriptor with just the one voter, and removing all else.